src/hypothesis/_settings.py

```diff
@@ -175,8 +176,28 @@ def defaults_stack(self):
             return self.storage.defaults_stack
 
     def __call__(self, test):
+        """Attach the settings as an attribute.
```

Comment:
:tada: yay docstring!  Not clear from the doc what it's attaching to, or why you would use it :wink:

Explain that, and a sentence or two at the end about the deprecation pathway will be easy (the current one only makes sense because I already know what it's saying).

Reply:
Uh, good point, oops. 😕